src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ChimericAlignment.java

```diff
@@ -59,89 +56,231 @@
+    static boolean nextAlignmentMayBeNovelInsertion(final AlignmentRegion current, final AlignmentRegion next,
+                                                    final Integer minAlignLength) {
+        return mapQualTooLow(next) ||                               // inserted sequence might have low mapping quality
+                firstAlignmentIsTooShort(next, current, minAlignLength) ||     // inserted sequence might be very small
+                current.referenceInterval.contains(next.referenceInterval) ||   // one might completely contain the other
+                next.referenceInterval.contains(current.referenceInterval);
+    }
 
     /**
      * Construct a new ChimericAlignment from two AlignmentRegions.
      * Assumes {@code regionWithLowerCoordOnContig} has a lower {@link AlignmentRegion#startInAssembledContig} than {@code regionWithHigherCoordOnContig}.
      */
-    protected ChimericAlignment(final AlignmentRegion regionWithLowerCoordOnContig, final AlignmentRegion regionWithHigherCoordOnContig,
-                                final String insertedSequence, final String homology,
-                                final List<String> insertionMappings) {
+    @VisibleForTesting
+    ChimericAlignment(final AlignmentRegion regionWithLowerCoordOnContig, final AlignmentRegion regionWithHigherCoordOnContig,
+                      final byte[] contigSequence, final List<String> insertionMappings,
+                      final SAMSequenceDictionary samSequenceDictionary) {
 
         final String assemblyId = regionWithLowerCoordOnContig.assemblyId;
         final String contigId = regionWithLowerCoordOnContig.contigId;
         Utils.validateArg(assemblyId.equals(regionWithHigherCoordOnContig.assemblyId), "two alignment regions used to construct chimeric alignment are not from the same local assembly.");
         Utils.validateArg(contigId.equals(regionWithHigherCoordOnContig.contigId), "two alignment regions used to construct chimeric alignment are not from the same assembled contig.");
 
-        this.assemblyId = assemblyId;
-        this.contigId = contigId;
         this.regionWithLowerCoordOnContig = regionWithLowerCoordOnContig;
         this.regionWithHigherCoordOnContig = regionWithHigherCoordOnContig;
-        this.homology = homology;
-        this.insertedSequence = insertedSequence;
+
+        this.strandSwitch = determineStrandSwitch(regionWithLowerCoordOnContig, regionWithHigherCoordOnContig);
+        this.involvesReferenceIntervalSwitch = involvesRefPositionSwitch(regionWithLowerCoordOnContig, regionWithHigherCoordOnContig, samSequenceDictionary);
+        this.isForwardStrandRepresentation = isForwardStrandRepresentation(regionWithLowerCoordOnContig, regionWithHigherCoordOnContig, this.strandSwitch, this.involvesReferenceIntervalSwitch);
+        this.isNotSimpleTranslocation = isNotSimpleTranslocation(regionWithLowerCoordOnContig, regionWithHigherCoordOnContig, this.strandSwitch, this.involvesReferenceIntervalSwitch);
```

Comment:
Not quite understanding the concerns, maybe in person explanation will be easier.